main/src/particle_store.F90

```diff
@@ -937,8 +937,8 @@ type (result_type) function particle_store_sample(pdata, psample, npmax, &
     end do
 
     ! Calculate sampling rate
-    fsample = min(1.0, real(npmax, kind=real8byte) / &
-         real(sum(np_area(1:nspecies)), kind=real8byte))
+    fsample = min(1.0_pos_kind, real(npmax, kind=pos_kind) / &
```

Comment:
maybe cast the whole thing into `real(kind=pos_kind)` instead of cast each number into pos_kind